geode-common/src/main/java/org/apache/geode/internal/inet/LocalHostUtil.java

```diff
@@ -54,70 +59,80 @@
   private static final boolean useLinkLocalAddresses =
       Boolean.getBoolean(USE_LINK_LOCAL_ADDRESSES_PROPERTY);
 
-  /**
-   * we cache localHost to avoid bug #40619, access-violation in native code
-   */
-  private static final InetAddress localHost;
-
   /**
    * all classes should use this variable to determine whether to use IPv4 or IPv6 addresses
    */
   @MakeNotStatic
   private static boolean useIPv6Addresses = !Boolean.getBoolean("java.net.preferIPv4Stack")
       && Boolean.getBoolean("java.net.preferIPv6Addresses");
 
-  static {
-    InetAddress inetAddress = null;
+  /**
+   * Resolves local host. Will retry if resolution fails.
+   *
+   * @return local host if resolved otherwise null.
+   */
+  private static InetAddress resolveLocalHost() {
```

Comment:
Could you add some type of test for the new "resolveLocalHost" method? If you had this method call a local method that then calls InetAddress.getLocalHost you could then mock that method and control when it would throw UnknownHostException to show that this new code will do retries.

Reply:
that method has to be a static method and we have to use powermock to mock a static method. We are trying to steer away from using power mock as far as I understand.

There are tests for Retry.

Reply:
If you want to test, you can create an overload that takes suppliers, functions, and predicates to call instead of calling static methods, then mock those things to drive the desired test conditions. Then `resolveLocalHost()` could simply call the overloaded one, passing references to the static methods, and we would leave that untested.

Reply:
There is a test that already asserts the retry behavior outside the context of the thing being retried. Testing this method specifically must be an integration test since it requires the use of a resolver and DNS. There are several tests that are tangentially dependent on this method given that nothing in Geode starts if this method returns `UnknownHostException`.